### PR TITLE
Feat(pipelines): Adds support for artifact nodes and vertical layout

### PR DESCRIPTION
**Pull request by jenny-s51**
## Description

https://issues.redhat.com/browse/RHOAIENG-3977

## How Has This Been Tested?

Tested locally using demo pipelines from backend team.

## Test Impact

As discussed cc @Gkrumbach07 , tests will be part of a standalone task: https://issues.redhat.com/browse/RHOAIENG-5146

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

Screenshot 2024-04-04 at 5 18 03 PM
Screenshot 2024-04-04 at 5 19 15 PM
Screenshot 2024-04-04 at 5 23 40 PM
Screenshot 2024-04-04 at 5 17 49 PM

After the PR is posted & before it merges:

- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
@jenny-s51 Can you log a testing task in jira, thanks!

**Comment by jenny-s51** [comment]
> @jenny-s51 Can you log a testing task in jira, thanks!

@Gkrumbach07  opened https://issues.redhat.com/browse/RHOAIENG-5146

**Review by jeff-phillips-18 — CHANGES_REQUESTED** [review]

File: frontend/src/concepts/topology/TaskEdge.tsx [hunk 1]
```diff
@@ -24,19 +25,21 @@ const TaskEdge: React.FunctionComponent<TaskEdgeProps> = ({
...
-        d={integralShapePath(startPoint, endPoint, startIndent, nodeSeparation)}
+        d={integralShapePath(startPoint, endPoint, startIndent, nodeSeparation, verticalLayout)}
         shapeRendering="geometricPrecision"
       />
 
       {element.getTarget().getType() !== DEFAULT_SPACER_NODE_TYPE ? (
         <ConnectorArrow
           className={styles.topologyEdge}
-          startPoint={endPoint.clone().translate(-1, 0)}
+          startPoint={endPoint.clone().translate(0, -1)}
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply frontend/src/concepts/topology/TaskEdge.tsx:42]: This is a good question @jeff-phillips-18 - are we making a full switch from horizontal to vertical pipelines moving forward @Gkrumbach07 @yannnz ?

File: frontend/src/concepts/topology/utils.ts [hunk 2]
```diff
@@ -5,6 +5,14 @@ import { NodeConstructDetails, PipelineNodeModelExpanded } from './types';
 
 export const createNodeId = (prefix = 'node'): string => `${prefix}-${genRandomChars()}`;
 
+export const ICON_TASK_NODE_TYPE = 'ICON_TASK_NODE';
+
+export const NODE_PADDING_VERTICAL = 0;
+export const NODE_PADDING_HORIZONTAL = 15;
+
+export const ARTIFACT_NODE_WIDTH = 56;
+export const ARTIFACT_NODE_HEIGHT = 43;
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply frontend/src/concepts/topology/utils.ts]: Nice catch, thank you! Makes more sense to just use `NODE_HEIGHT` here.

File: frontend/src/concepts/topology/TaskEdge.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply frontend/src/concepts/topology/TaskEdge.tsx:42]: I dont see why we would go back to horizontal. We should be able to remove it

**Review by jeff-phillips-18 — APPROVED** [review]
/lgtm

**Comment by andrewballantyne** [comment]
LGTM code wise -- a lot of the changes were in Pipelines Core and I think that was reviewed by Phillips well. Very low foot print in our world.

I have not had a chance to test this, just did a sweep of the code as @Gkrumbach07 asked.

**Comment by Gkrumbach07** [comment]
Tests are passing. I tested on as many graphs as I could. This is fairly low impact as its mostly changing styling and not too much parsing. Tests are logged for later

/lgtm
/approve

**Pull request merged**